lib/offsite_payments/integrations/paytm.rb

```diff
@@ -0,0 +1,475 @@
+module OffsitePayments #:nodoc:
+  module Integrations #:nodoc:
+    module Paytm
+	
+	  require 'openssl'
+	  require 'base64'
+	  require 'digest'
+	  require 'securerandom'
+	
+	def new_pg_encrypt(params)
+		if (params.class != Hash) || (params.keys == [])
+		  return false
+		end
+		if !params.has_key?(:key)
+		  return false
+		end
+		encrypted_data = Hash[]
+		key = params.delete(:key)
+		keys = params.keys
+		aes = OpenSSL::Cipher::Cipher.new("aes-128-cbc")
+		begin
+		  keys.each do |k|
+			data = params[k]
+			aes.encrypt
+			aes.key = key
+			aes.iv = '@@@@&&&&####$$$$'
+			encrypted_k = aes.update(k.to_s) + aes.final
+			encrypted_k = Base64.encode64(encrypted_k.to_s)
+			aes.encrypt
+			aes.key = key
+			aes.iv = '@@@@&&&&####$$$$'
+			encrypted_data[encrypted_k] = aes.update(data.to_s) + aes.final
+			encrypted_data[encrypted_k] = Base64.encode64(encrypted_data[encrypted_k])
+		  end
+		rescue Exception => e
+		  return false
+		end
+		return encrypted_data
+	end
+  
+  
+	  def new_pg_encrypt_variable(data, key)
+		aes = OpenSSL::Cipher::Cipher.new("aes-128-cbc")
```

Comment:
The logic of instantiating a cipher, setting the key, iv and creating the encrypted data is repeated quite frequently. Could this logic be moved to a single method to avoid repeating the code so many times?